model-optimizer/mo/front/kaldi/loader/loader.py

```diff
@@ -44,52 +43,52 @@ def load_parallel_component(file_descr, graph: Graph, prev_layer_id):
     nnet_count = read_token_value(file_descr, b'<NestedNnetCount>')
     log.debug('Model contains parallel component with {} nested networks'.format(nnet_count))
 
-    slice_id = graph.unique_id(prefix='Slice')
-    graph.add_node(slice_id, parameters=None, op='slice', kind='op')
-
-    slice_node = Node(graph, slice_id)
-    Node(graph, prev_layer_id).add_output_port(0)
-    slice_node.add_input_port(0)
-    graph.create_edge(Node(graph, prev_layer_id), slice_node, 0, 0)
-    slices_points = []
-
+    split_points = []
     outputs = []
+    inputs = []
 
     for i in range(nnet_count):
         read_token_value(file_descr, b'<NestedNnet>')
         collect_until_token(file_descr, b'<Nnet>')
         g = Graph()
         load_kalid_nnet1_model(g, file_descr, 'Nested_net_{}'.format(i))
-        input_nodes = [n for n in graph.nodes(data=True) if n[1]['op'] == 'Parameter']
-        shape = input_nodes[0][1]['shape']
-        if i != nnet_count - 1:
-            slices_points.append(shape[1])
-        g.remove_node(input_nodes[0][0])
+
+        input_nodes = [n for n in g.nodes(data=True) if n[1]['op'] == 'Parameter']
+        if len(input_nodes) == 1:
+            shape = input_nodes[0][1]['shape']
+            split_points.append(shape[1])
```

Comment:
Could you provide some comment in the code why do we use this value as a split points?

Reply:
Added comment in the new version